var/spack/repos/builtin/packages/root/package.py

```diff
@@ -326,6 +326,79 @@ def determine_version(cls, exe):
+        _add_variant_from_feature(variants, features, 'dcache', '+dcache')
+        _add_variant_from_feature(variants, features, 'fftw3', '+fftw')
+        _add_variant_from_feature(variants, features, 'fitsio', '+fits')
+        _add_variant_from_feature(variants, features, ('ftgl', 'opengl'), '+opengl')
+        _add_variant_from_feature(variants, features, 'gdml', '+gdml')
+        _add_variant_from_feature(variants, features, 'mathmore', '+math')
+        _add_variant_from_feature(variants, features, 'gviz', '+graphviz')
+        _add_variant_from_feature(variants, features, 'http', '+http')
+        _add_variant_from_feature(variants, features, ('imt', 'tbb'), '+tbb')
+        _add_variant_from_feature(variants, features, 'jemalloc', '+jemalloc')
+        _add_variant_from_feature(variants, features, 'memstat', '+memstat')
+        _add_variant_from_feature(variants, features, ('minuit', 'minuit2'), '+minuit')
+        _add_variant_from_feature(variants, features, 'mlp', '+mlp')
+        _add_variant_from_feature(variants, features, 'mysql', '+mysql')
+        _add_variant_from_feature(variants, features, 'oracle', '+oracle')
+        _add_variant_from_feature(variants, features, 'pgsql', '+postgres')
+        _add_variant_from_feature(variants, features, 'pythia6', '+pythia6')
+        _add_variant_from_feature(variants, features, 'pythia8', '+pythia8')
+        _add_variant_from_feature(variants, features, 'pyroot', '+python')
+        _add_variant_from_feature(variants, features, ('qt', 'qtgsi'), '+qt4')
+        _add_variant_from_feature(variants, features, 'r', '+r')
+        _add_variant_from_feature(variants, features, 'roofit', '+roofit')
+        _add_variant_from_feature(variants, features, ('root7', 'webui'), '+root7')
+        _add_variant_from_feature(variants, features, 'rpath', '+rpath')
+        _add_variant_from_feature(variants, features, 'shadowpw', '+shadow')
+        _add_variant_from_feature(variants, features, 'spectrum', '+spectrum')
+        _add_variant_from_feature(variants, features, 'sqlite', '+sqlite')
+        _add_variant_from_feature(variants, features, 'ssl', '+ssl')
+        _add_variant_from_feature(variants, features, 'table', '+table')
+        _add_variant_from_feature(variants, features, 'thread', '+threads')
+        _add_variant_from_feature(variants, features, 'tmva', '+tmva')
+        _add_variant_from_feature(variants, features, 'unuran', '+unuran')
+        _add_variant_from_feature(variants, features, 'vc', '+vc')
+        _add_variant_from_feature(variants, features, 'vdt', '+vdt')
+        _add_variant_from_feature(variants, features, 'veccore', '+veccore')
+        _add_variant_from_feature(variants, features, 'vmc', '+vmc')
+        _add_variant_from_feature(variants, features, ('x11', 'xft'), '+x')
+        _add_variant_from_feature(variants, features, 'xml', '+xml')
+        _add_variant_from_feature(variants, features, 'xrootd', '+xrootd')
```

Comment:
There's a lot of duplication here. I would use a dict or list and loop over the entries instead

Reply:
Hmmm, I'm not sure that makes a lot of difference. In the end, I need to write out the variant and feature names, and if I put that in a list or one macro-like function call per line is basically the same. In the future maybe we could refactor things to have one list of variants  and build flags in the recipe  that can be used in different places.

I did think about adding a `nonlocal` to `add_variant_from_feature`, to eliminate the first two arguments, but that doesn't work with python 2 I think. So as a compromise I just choose some more concise names for the repeated things.